optimum/intel/openvino/utils.py

```diff
@@ -228,3 +232,275 @@ def model_has_dynamic_inputs(model):
+            err.filename = path
+            onexc(func, path, err)
+
+    def _rmtree_unsafe(path, onexc):
+        def onerror(err):
+            onexc(os.scandir, err.filename, err)
+
+        results = os.walk(path, topdown=False, onerror=onerror, followlinks=os._walk_symlinks_as_files)
+        for dirpath, dirnames, filenames in results:
+            for name in dirnames:
+                fullname = os.path.join(dirpath, name)
+                try:
+                    os.rmdir(fullname)
+                except OSError as err:
+                    onexc(os.rmdir, fullname, err)
+            for name in filenames:
+                fullname = os.path.join(dirpath, name)
+                try:
+                    os.unlink(fullname)
+                except OSError as err:
+                    onexc(os.unlink, fullname, err)
+        try:
+            os.rmdir(path)
+        except OSError as err:
+            onexc(os.rmdir, path, err)
+
+    if ignore_errors:
+
+        def onexc(*args):
+            pass
+
+    elif onerror is None and onexc is None:
+
+        def onexc(*args):
+            raise
+
+    elif onexc is None:
+        if onerror is None:
+
+            def onexc(*args):
+                raise
+
+        else:
+            # delegate to onerror
+            def onexc(*args):
+                func, path, exc = args
+                if exc is None:
+                    exc_info = None, None, None
+                else:
+                    exc_info = type(exc), exc, exc.__traceback__
+                return onerror(func, path, exc_info)
+
+    if _use_fd_functions:
+        # While the unsafe rmtree works fine on bytes, the fd based does not.
+        if isinstance(path, bytes):
+            path = os.fsdecode(path)
+        stack = [(os.lstat, dir_fd, path, None)]
+        try:
+            while stack:
+                _rmtree_safe_fd(stack, onexc)
+        finally:
+            # Close any file descriptors still on the stack.
+            while stack:
+                func, fd, path, entry = stack.pop()
+                if func is not os.close:
+                    continue
+                try:
+                    os.close(fd)
+                except OSError as err:
+                    onexc(os.close, path, err)
+    else:
+        if dir_fd is not None:
+            raise NotImplementedError("dir_fd unavailable on this platform")
+        try:
+            if _rmtree_islink(path):
+                # symlinks to directories are forbidden, see bug #1669
+                raise OSError("Cannot call rmtree on a symbolic link")
+        except OSError as err:
+            onexc(os.path.islink, path, err)
+            # can't continue even if onexc hook returns
+            return
+        return _rmtree_unsafe(path, onexc)
+
+
+# copied https://github.com/python/cpython/blob/3.12/Lib/tempfile.py
+# to add behaviour that available only for python3.10+ for older python version
+class TemporaryDirectory(OrigTemporaryDirectory):
```

Comment:
@eaidova , could you please also replace `tempfile.TemporaryDirectory` with this custom one in (all?) openvino tests?

For example, I believe the test below fails because of this.
```
FAILED tests/openvino/test_quantization.py::OVQuantizerQATest::test_automodel_static_quantization_0_hf_internal_testing_tiny_random_BertForQuestionAnswering - PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmp_q_68_5j\\openvino_model.bin'
```

Reply:
ok, added